dateutil/parser/_parser.py

```diff
@@ -963,7 +939,7 @@ def _parse_numeric_token(self, tokens, idx, info, ymd, res, fuzzy):
                     if value is not None:
                         ymd.append(value, 'M')
                     else:
-                        raise InvalidDatetimeError(ymd.tzstr)
+                        raise InvalidDatetimeError()
```

Comment:
Admittedly accessing the input via ymd.tzstr was a bit of a kludge, but it still seems like it should be included in the exception msg.  Maybe attach it to `res` since that is kind of a ParseState?

While we're at it, maybe truncate the string that goes in there at, say, 128 characters?

Reply:
No need for any of that. For one thing, none of these exceptions ever made it out of `_parse`, because they all got caught and raised as `ValueError` for whatever reason.

If we want to fix that as part of the cleanup, we can just catch, attach the string in the `_parse` scope, then re-raise.